benchmarking/simulation/simulation.py

```diff
@@ -50,6 +53,13 @@ def run(self, num_runs: int) -> SimulationArtifact:
                 else:
                     num_runs -= len(cached_team_sets)
 
+        completed_run_indexes = []
+        for team_set in self.team_sets:
+            team_set_id = team_set._id
+            match = re.search(r"run_index_(\d+)", team_set_id)
```

Comment:
we need to be like pretty confident we don't break this id naming convention

Reply:
I'd feel more comfortable if this was a number just placed somewhere.

I was envisioning more like we save the used run indexes in the metadata of the cache file

And then merge those back together in the combine function

And then we can just get the list of used run indexes all at once